src/LogViewer.php

```diff
@@ -0,0 +1,211 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+use Glpi\Application\View\TemplateRenderer;
+use Glpi\Event;
+
+class LogViewer extends CommonGLPI
+{
+    public static $rightname = 'logs';
+
+
+    public static function getTypeName($nb = 0)
+    {
+        return _n('Log', 'Logs', $nb);
+    }
+
+
+    public static function getMenuContent()
+    {
+        $menu = [];
+
+        if (self::canView()) {
+            $menu = [
+                'title'   => self::getTypeName(Session::getPluralNumber()),
+                'page'    => '/front/logs.php',
+                'icon'    => static::getIcon(),
+                'options' => [
+                    'logfile' => [
+                        'title' => __('Log file'),
+                        'page'  => '/front/logviewer.php',
+                        'icon'  => 'ti ti-file',
+                    ],
+                ],
+            ];
+
+            if (Event::canView()) {
+                $menu['options']['Glpi\\Event'] = [
+                    'title' => Event::getTypeName(Session::getPluralNumber()),
+                    'page'  => Event::getSearchURL(false),
+                    'icon'  => Event::getIcon(),
+                ];
+            }
+        }
+
+        return $menu;
+    }
+
+
+    /**
+     * Display a link for events and a list of log files links
+     *
+     * @return void
+     */
+    public function displayList()
+    {
+        TemplateRenderer::getInstance()->display(
+            'pages/admin/logs_list.html.twig',
+            [
+                'logs' => self::getLogsFilesList(),
+            ]
+        );
+    }
+
+
+    /** get a list of log files of GLPI
+     *
+     * @return array of [slug => filename.log]
+     */
+    protected static function getLogsFilesList(): array
+    {
+        $logs_files = scandir(GLPI_LOG_DIR);
+        $files = [];
+        foreach ($logs_files as $log_filename) {
+            if (preg_match('/^(.+)\.log$/', $log_filename, $matches)) {
+                $filename = $matches[1];
+                $filekey  = Toolbox::slugify($filename, '', true);
+                $files[$filekey] = "$filename.log";
+            }
+        }
+        return $files;
+    }
+
+
+    /** Display a log file content
+     *  We try to explode the log file by searching datetime patterns
+     *
+     * @param string $fileslug the slug of the log file
+     * @param int $nb_lines
+     *
+     * @return void
+     */
+    public static function showLogFile(string $fileslug = "", bool $only_content = false, int $nb_lines = null)
+    {
+        global $CFG_GLPI;
+
+        $logfiles = self::getLogsFilesList();
+        $filename = $logfiles[$fileslug] ?? "";
+        $filepath = GLPI_LOG_DIR . "/" . $filename;
+        if (!file_exists($filepath)) {
```

Comment:
```suggestion
        if (is_dir($filepath) || !file_exists($filepath)) {
```
An invalid log name would make this code try reading from the log directory.